### ISSUE TITLE
Tiny ants / wasps

### ISSUE THREAD

**Issue by attila123** [colobot/colobot#610]
I think this problem comes from the original colobot game. E.g. in the user level pack "Jinx Incident" I could see very small wasps under certain circumstances.
Now it happened in Terranova 4 level somehow (playing with latest dev).
Savegame (for the screenshot below):
https://www.dropbox.com/s/3redpfap6yxvbg0/save1508290251Terranova4.zip
These small ants / wasps are more difficult to notice and impossible to kill (see below), but they do equal amount of damage, I guess. So they are quite dangerous actually. And very frustrating if they appear...
Towers do not react to them. I tried to shoot one with an orga shooter, then with Phazer, no effect. This screenshot shows a tiny ant (look at the left track of the shielder).

Please implement at least workaround so that at loading these tiny insects get back to normal size (zoom=1;1;1), and is possible to kill them (manually correcting the zoom does not make them vulnerable).

![tiny_ant](https://cloud.githubusercontent.com/assets/1728046/9559726/a9643652-4df8-11e5-82f2-c88d1c323d5d.png)

**Comment by krzys-h** [comment]
Is this after saving and loading or not? They are probably somehow exiting eggs too early.
If you are looking for a manual fix, there should be lock=1 somewhere in CreateObject, change that to 0

**Comment by attila123** [comment]
Yes, I think it's connected with saving and loading. If I remember correcly in Jinx Incident before destroying the alien nest with Phazer. I saved the game and when loading it all the nearby eggs became tiny wasps.
I think modifying a level placing an alien egg next to the starting position could help to reproduce it.
Thanks, the manual fix worked, lock=1 was consistent with the small ants.

**Comment by attila123** [comment]
I managed to reproduce it. Add a new program to e.g. a winged bot (practical as you can get a bit further before ants start to hatch):

produce(position, orientation, AlienAnt, "antt41.txt");
produce(position, orientation, AlienAnt, "antt42.txt");
produce(position, orientation, AlienAnt, "antt43.txt");

While flying (to get to a distance from eggs), click on this program a lot of times.
Save it quickly e.g. when the first eggs start to hatch.
In my first savegame there were 61 ants (maybe one was created before by the AlienQueen), one of them with lock=1.
Then I did 136, and 20 of them was saved with lock=1.

Of course this could be done with any other bot (don't have to survive the ants) or with the AlienQueen, whatever is easier.
This also stresses the sound system nicely (see #615 ).

**Comment by krzys-h** [comment]
lock=1 actually should happen while the alien is inside of the egg, but then the program should not run. It might be that the active= property is not saved

**Comment by krzys-h** [comment]
Please try now. Keep in mind this will only work for newly created objects, not for wrongly saved ones.

**Comment by attila123** [comment]
"lock=1 actually should happen while the alien is inside of the egg". ?Hmm, then when I thought that I reproduced it, I was wrong. :(
Also my method created a lot of ants in the same place, which causes other side effects too: these ants can't move at all, or they try to go to the same direction (now I tried it on Moon 1), but since they overlap each other, they move quite slow. Or after loading some eggs could not even hatch because of the overlapping (the ants in them collide with each other?). -> maybe it was with the old version.
So this is difficult to test...
A correct script should be written which would produce 1-2 hundred ants but not overlapping each other and created slightly delayed. They would need to run some scripts, but better not firing.
Then one would need to save during hatching + load that game and wait for all hatched and save again, check for lock=1 in the second save. This method should be tested with the old version (before the activity change fix), if this is good enough method to reproduce.
Then test again with the new version.
So it's getting quite complicated. I'm giving up now... :(
